test/testsuites/migrationtests.go

```diff
@@ -0,0 +1,60 @@
+package testsuites
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	olm "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+	"github.com/operator-framework/operator-marketplace/test/helpers"
+	"github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// MigrationTests is a test suite that ensures the following:
+// * All stale datastore CatalogSourceConfigs created by marketplace after the creation of
+// OperatorSources in a 4.1.x cluster are deleted when the cluster is migrated to openshift 4.2.0.
+// * Installed CatalogSourceConfigs created during operator installation in a 4.1 cluster are deleted.
+// * Existing Subscriptions are updated to reference the datastore CatalogSources instead
+// of installed CatalogSources created during operator installation in a openshift 4.1.x cluster.
+func MigrationTests(t *testing.T) {
+	t.Run("catalogsourceconfigs-are-cleaned-up", testCatalogSourceConfigsCleanedUp)
+	t.Run("subscriptions-are-updated", testSubscriptionsUpdated)
+}
+
+// testCatalogSourceConfigsCleanedUp ensures that after a cluster is migrated
+// from openshift 4.1.x to openshift 4.2.0, the following stale objects are cleaned up:
+// Stale CatalogSourceConfigs created during operator installation in a openshift 4.1.x cluster.
+// Datastore CatalogSourceConfigs created by OperatorSources in a openshift 4.1.x cluster.
+func testCatalogSourceConfigsCleanedUp(t *testing.T) {
+	// Get test namespace
+	namespace, err := test.NewTestCtx(t).GetNamespace()
```

Comment:
Don't you need to `defer ctx.Cleanup()` to be on the safe side?

Reply:
Isn't it done from the parent testgroup here https://github.com/operator-framework/operator-marketplace/pull/223/files/5c068a0e5aef87360da32e2b7e7288ac797109ab#diff-33a154e1b96c5e7d9d61ed336d00d7efR20? Maybe I misunderstood

Reply:
You are creating a new context here.

Reply:
I see - thanks!